src/boot/primary.js

```diff
@@ -0,0 +1,222 @@
+        const sessionExpirationTime = parseInt(store.state.account.createdAt) + parseInt(store.state.account.expiresIn);
+
+        // console.info('createdAt', Object.prototype.toString.call(store.state.account.createdAt), 'expiresIn', Object.prototype.toString.call(store.state.account.expiresIn))
+        // console.info('currentTimestamp', currentTimestamp, Object.prototype.toString.call(currentTimestamp))
+        // console.info('sessionExpirationTime', sessionExpirationTime, Object.prototype.toString.call(sessionExpirationTime))
+
+        if (currentTimestamp > sessionExpirationTime ||
+            (currentTimestamp - store.state.globalConsts.SAFETY_BEFORE_SESSION_TIMEOUT) > sessionExpirationTime) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    function attemptRefreshToken () {
+        return new Promise(function (resolve, reject) {
+            window.$.ajax({
+                type: 'POST',
+                url: store.state.metaverseConfig.server + '/oauth/token',
+                contentType: 'application/x-www-form-urlencoded;charset=UTF-8',
+                data: {
+                    grant_type: 'refresh_token',
+                    scope: store.state.account.scope,
+                    refresh_token: store.state.account.refreshToken
+                }
+            })
+                .done(function (result) {
+                    store.commit('mutate', {
+                        update: true,
+                        property: 'account',
+                        with: {
+                            isLoggedIn: true,
+                            accessToken: result.access_token,
+                            tokenType: result.token_type,
+                            createdAt: result.created_at,
+                            expiresIn: result.expires_in,
+                            refreshToken: result.refresh_token,
+                            scope: result.scope
+                        }
+                    });
+                    Log.print('METAVERSE', 'INFO', 'Token refresh successful.');
+                    resolve();
+                })
+                .fail(function (result) {
+                    // If this fails for any reason, the user must log back in.
+                    Log.print('METAVERSE', 'WARN', 'Refresh failed.');
+                    store.state.Metaverse.logout();
+                    reject(result.responseJSON || result.responseText);
+                });
+        });
+    }
+
+    function parseFromStorage (item) {
+        let itemToSendBack;
+        const retrievedItem = localStorage.getItem(item);
+
+        if (!retrievedItem || retrievedItem === 'undefined') {
+            itemToSendBack = undefined;
+        } else {
+            try {
+                itemToSendBack = JSON.parse(retrievedItem);
+            } catch (error) {
+                console.info('Error retrieving', item, 'from storage. Parsing error:', error);
+            }
+        }
```

Comment:
```suggestion
        let itemToSendBack = undefined;
        const retrievedItem = localStorage.getItem(item);
        if (retrievedItem && retrievedItem !== 'undefined') {
            try {
                itemToSendBack = JSON.parse(retrievedItem);
            } catch (error) {
                console.info('Error retrieving', item, 'from storage. Parsing error:', error);
            }
        }
```
Also, does `localStorage.getItem()` really return a string, `'undefined'`? Or does it return a value of `undefined`?

Reply:
- if getItem() is really returning the string 'undefined' then I would request this... not happen.  Would feel like an inappropriate use of toString() to accidentally get that.
- the value undefined is falsy so "if(retrievedItem)" would filter it out.